### PR TITLE
feat: support masked_scatter by lowering path

### PR DESCRIPTION
**Pull request by chohk88**
# Description

Implemented support for `masked_scatter` in the lowering path, referring to [this implementation in PyTorch Inductor](https://github.com/pytorch/pytorch/blob/924a247fbbe32063966533535b1cc0cf474ef1ed/torch/_inductor/decomposition.py#L831).

Fixes # ([issue](https://github.com/pytorch/TensorRT/issues/3410))

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 1]
```diff
@@ -566,6 +566,46 @@ def scaled_dot_product_cudnn_attention_decomposition(
...
+    source_idx = mask_flat.to(torch.int64).cumsum(0) - 1
+
+    # 4) Gather elements from source_flat using these indices
+    gathered = source_flat.gather(0, source_idx)
+
+    # 5) Replace positions where mask is True with gathered values, otherwise keep original
+    replaced = torch.where(mask_flat, gathered, input_flat)
+
+    # 6) Reshape the result back to the broadcasted shape
+    return replaced.view(input_b.shape)
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:606]: @chohk88  I have a question. I tried running this code in a separate python session for the test input size of (2, 3, 4) and I see the following error. Do you know why this happens ? Am I missing something here ? 
```py
import torch
shape=(2, 3, 4)

ax=torch.randn(*shape, dtype=torch.float32, device="cuda")
mask=torch.rand(*shape, device="cuda") > 0.5
num_trues = mask.sum().item()
source = torch.arange(num_trues, dtype=torch.float32, device="cuda")
ax_b, mask_b = torch.ops.aten.broadcast_tensors([ax, mask])
ax_flat = ax_b.flatten()
mask_flat = mask_b.flatten()
source_flat = source.flatten()
source_idx = mask_flat.to(torch.int64).cumsum(0) - 1
gathered = source_flat.gather(0, source_idx)
>>> /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:144: operator(): block: [0,0,0], thread: [0,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
/pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:144: operator(): block: [0,0,0], thread: [1,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
```

**Comment by chohk88** [comment]
Closing this PR due to an incorrect author. Reopened as [PR](https://github.com/pytorch/TensorRT/pull/3476)
